packages/bsp/common/etc/update-motd.d/10-armbian-header

```diff
@@ -15,21 +15,46 @@ MOTD_DISABLE=""
 [[ -f /etc/default/armbian-motd ]] && . /etc/default/armbian-motd
 
 for f in $MOTD_DISABLE; do
-	[[ $f == $THIS_SCRIPT ]] && exit 0
+		[[ $f == $THIS_SCRIPT ]] && exit 0
 done
 
 . /etc/os-release
-. /etc/armbian-release
 
+# extract code name only
+PRETTY_NAME=$(sed 's/.*(\(.*\)).*/\1/' <<< "$PRETTY_NAME" | awk '{print $1}' | sed 's/.*/\u&/')
+
+# read kernel version
 KERNELID=$(uname -r)
 
-PRETTY_NAME=$(sed 's/.*(\(.*\)).*/\1/' <<< "$PRETTY_NAME" | awk '{print $1}' | awk '{print tolower($0)}')
-[[ "$IMAGE_TYPE" != "nightly" && "$VERSION" == 20* ]] && VERSION=$(echo $VERSION | cut -f1,2 -d'.')
+# read image data
+. /etc/armbian-image-release
+IMAGE_TYPE_INITIAL=$IMAGE_TYPE
+
+# read updated data
+. /etc/armbian-release
+
+# show commit info if exists otherwise version
+if [[ -n $BUILD_REPOSITORY_COMMIT ]]; then
+		SHOW_VERSION="\e[0;91m"$BUILD_REPOSITORY_COMMIT"\x1B[0m"
+else
+		SHOW_VERSION="\e[0;92m$VERSION\x1B[0m"
+fi
+
+# stable versions
+if [[ $IMAGE_TYPE_INITIAL == $IMAGE_TYPE && $IMAGE_TYPE == stable && $BUILD_REPOSITORY_URL == "https://github.com/armbian/build" ]]; then
+		if [[ "$VERSION" == 20* ]]; then
+		# show only prefix
+		SHOW_VERSION="\e[0;92m"$(echo $VERSION | cut -f1,2 -d'.')"\x1B[0m"
+		else
+		SHOW_VERSION="\e[0;92m$VERSION\x1B[0m"
+		fi
+fi
+
 
 if [ $(echo $BOARD_NAME | wc -c) -ge 18 ]; then
-	TERM=linux toilet -f standard -F metal $(echo $BOARD_NAME | sed 's/Orange Pi/OPi/' | sed 's/Nanopi/NPi/')
+		TERM=linux toilet -f standard -F metal $(echo $BOARD_NAME | sed 's/Orange Pi/OPi/' | sed 's/Nanopi/NPi/')
 else
-	TERM=linux toilet -f standard -F metal $BOARD_NAME
+		TERM=linux toilet -f standard -F metal $BOARD_NAME
 fi
 
-echo -e "\nWelcome to \e[0;91mArmbian $PRETTY_NAME $VERSION\x1B[0m $IMAGE_TYPE $KERNELID"
+echo -e "Welcome to \e[0;91mArmbian\x1B[0m - $PRETTY_NAME\x1B[0m ($SHOW_VERSION) Linux $KERNELID\n"
```

Comment:
Ubuntu-/Debian-version is now missing (e.g. 18.04 or Bionic). I would prefer having it still there.

Reply:
Aha, I see. There it should be written Bionic not Ubuntu. Fixing ...

Reply:
```
 _   _                         _   __  __ _  _   
| \ | | __ _ _ __   ___  _ __ (_) |  \/  | || |  
|  \| |/ _` | '_ \ / _ \| '_ \| | | |\/| | || |_ 
| |\  | (_| | | | | (_) | |_) | | | |  | |__   _|
|_| \_|\__,_|_| |_|\___/| .__/|_| |_|  |_|  |_|  
                        |_|                      
Welcome to Armbian - Xenial (2018.10) Linux 4.4.161-rk3399`

Reply:
Great! `Armbian - Xenial (2018.10)` -- so as usual we babble a lot, express concerns [here](https://github.com/armbian/build/pull/1129#pullrequestreview-163752099
), [here](https://github.com/armbian/build/pull/1129#issuecomment-429015599) and [there](https://github.com/armbian/build/pull/1129#issuecomment-429079124) and all that's happening is ignoring everything and doing your own thing. Well done.